### PR TITLE
Use `proto.Clone` to Deep-Copy `proto.Message`s

### PR DESCRIPTION
**Pull request by t0rr3sp3dr0**
The implementation of this project uses `json.Marshal` and `json.Unmarshal` to deep-copy the `proto.Message` before redacting sensitive fields. This may lead to unwanted effects depending of the JSON tags the struct has and whether or not it implements `json.Marshaler` and `json.Unmarshaler`. This patch uses `proto.Clone` to deep-copy `proto.Message`s instead of the `encoding/json` functions.

I also reworked the tests so you can just run `make test` to test everything, even after changing the String function implementation.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -15,3 +15,9 @@ bin/**
 
 # Don't checkin the binary
 grpc-go-redact
+
```
Comment (bfoley13) [inline .gitignore:18]: just delete the files from the repo

File: Makefile [hunk 2]
```diff
@@ -1,13 +1,28 @@
+all: clean build test
+.PHONY: all
 
+clean:
+	rm -fR ./bin/*
+	rm -fR ./grpc-go-redact
+	rm -fR ./test/input.pb.go
```
Comment (bfoley13) [inline Makefile:7]: remove this one you delete them

File: .gitignore (hunk 1 above)
Comment (bfoley13) [reply .gitignore:18]: actually youre trying to regenerate them in the tests now? we shouldnt have make commands that reference files made within tests?

Comment (t0rr3sp3dr0) [reply .gitignore:18]: These files are being generated when you run `make test`. Previsouly, if you changed the String func implementation, you had to manually regenerate test files and it was not clear how to do that. Now you just run `make test` and the Makefile will hadle everything for you.

File: Makefile [hunk 3]
```diff
@@ -1,13 +1,28 @@
...
-	./bin/grpc-go-redact -input ./test/base/test.pb.go -output ./test/output.pb.go
+test: test/output.pb.go
+	go test ./...
+.PHONY: test
+
+bin/grpc-go-redact:
+	go build -o ./bin/grpc-go-redact .
 
+test/input.txt:
+	sed -E -e 's/^package internal$$/package test/g' -e '/^(\/\*|\*\/)$$/d' ./generator/internal/stringfunc.pb.go > ./test/input.txt
```
Comment (bfoley13) [inline Makefile:24]: totally missed this

**Review by bfoley13 — APPROVED** [review]

**Pull request merged**